---
title: Microbiome - Bioconductor/curatedMetagenomicData meta analysis
author: 'Dror Berel'
output: BiocStyle::html_document
vignette: |-
  %\VignetteIndexEntry{Test}
                             %\VignetteEngine{knitr::rmarkdown}
                             \usepackage[utf8]{inputenc}
---


```{r setup, include=FALSE}

#packman will check if these packages are installed and install them if not.
if (!require("pacman")) install.packages("pacman")
pacman::p_load(knitr, limma, dplyr, DT,tibble, tidyr, purrr, here, ggplot2, ggfortify, magritter, SummarizedExperiment, biobroom, kableExtra, corrplot, htmltools, ComplexHeatmap, gridExtra, pheatmap, formattable, GGally, stringr, curatedMetagenomicData, phyloseq)


knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(limma)
library(dplyr)
library(DT)
library(tibble); options(tibble.width = Inf)
library(tidyr) 
library(purrr)
library(here); here()
library(ggplot2)
library(ggfortify)
library(magrittr)
library(SummarizedExperiment)
library(biobroom) # tidy() # not to confuse with broom::tidy !!!
library(kableExtra)
library(corrplot)
library(htmltools)

library(ComplexHeatmap)
library(gridExtra)
library(pheatmap)
library(formattable)
library(GGally)
library(stringr)


library(curatedMetagenomicData)
library(phyloseq)


# https://htmlpreview.github.io/?https://github.com/NCBI-Hackathons/Curated_Metagenomics/blob/master/curatedMetagenomicData_data_review.html

```



# Bioconductir data package:

(http://bioconductor.org/packages/release/data/experiment/html/curatedMetagenomicData.html)

```{r}
if(as.numeric(version$minor) >= 5 & as.numeric(version$major >= 3)) {
  if (!requireNamespace("BiocManager", quietly = TRUE))
      install.packages("BiocManager")
  BiocManager::install("curatedMetagenomicData", version = "3.8")
}
```

(http://bioconductor.org/packages/release/data/experiment/vignettes/curatedMetagenomicData/inst/doc/curatedMetagenomicData.html)


# Paper:
(https://www.nature.com/articles/nmeth.4468)



# 6 types of data for each dataset:  
    1. Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level  
    2. Presence of unique, clade-specific markers  
    3. Abundance of unique, clade-specific markers  
    4. Abundance of gene families  
    5. Metabolic pathway coverage  
    6. Metabolic pathway abundance  

Types 1-3 are generated by MetaPhlAn2; 4-6 are generated by HUMAnN2.  




# Taxanomies: 
rank_names( loman.pseq ):   
 Kingdom  
 Phylum  
 Class  
 Order  
 Family  
 Genus  
 Species  
 Strain  


 Taxonomy Table:     [6 taxa by 8 taxonomic ranks]:  
                   Kingdom    Phylum           Class         Order Family  
 k__Bacteria       "Bacteria" NA               NA            NA    NA      
 p__Bacteroidetes  "Bacteria" "Bacteroidetes"  NA            NA    NA    
 p__Firmicutes     "Bacteria" "Firmicutes"     NA            NA    NA    
 p__Proteobacteria "Bacteria" "Proteobacteria" NA            NA    NA    
 c__Bacteroidia    "Bacteria" "Bacteroidetes"  "Bacteroidia" NA    NA    
 c__Clostridia     "Bacteria" "Firmicutes"     "Clostridia"  NA    NA    
                   Genus Species Strain
 k__Bacteria       NA    NA      NA    
 p__Bacteroidetes  NA    NA      NA    
 p__Firmicutes     NA    NA      NA    
 p__Proteobacteria NA    NA      NA    
 c__Bacteroidia    NA    NA      NA    
 c__Clostridia     NA    NA      NA



# Goals 
## Test the potential in public data repository for microbiome dataset   

31 multi-assay studies, each with multiple assays
```{r ,  warning = FALSE, message = FALSE}
ES_studies_tib<-combined_metadata %>%
  as.data.frame() %>% 
  group_by(dataset_name) %>% nest(.key = 'Study')

ES_studies_tib$dataset_name

cat("study's characteristics")
ES_studies_tib$Study[[1]] %>% glimpse

# LomanNJ_2013
LomanNJ_1<-LomanNJ_2013.genefamilies_relab.stool()
LomanNJ_2<-LomanNJ_2013.marker_abundance.stool()
LomanNJ_3<-LomanNJ_2013.marker_presence.stool()
LomanNJ_4<-LomanNJ_2013.metaphlan_bugs_list.stool()
LomanNJ_5<-LomanNJ_2013.pathabundance_relab.stool()
LomanNJ_6<-LomanNJ_2013.pathcoverage.stool()

## LomanNJ from curatedMetagenomicData
LomanNJ_tib<-list(genefamilies_relab = LomanNJ_1, 
                  marker_abundance = LomanNJ_2, 
                  marker_presence = LomanNJ_3, 
                  metaphlan_bugs_list = LomanNJ_4, 
                  pathabundance_relab = LomanNJ_5, 
                  pathcoverage = LomanNJ_6) %>% enframe('type','ES')

LomanNJ_tib %<>% 
  mutate( dim =        ES %>% map(~dim(.x))) %>% 
  mutate( n_Samples= dim %>% map_int('Samples')) %>% 
  mutate( n_Features= dim %>% map_int('Features')) %>% 
  mutate( exprs_head = ES %>% map(~exprs(.x)[1:5, 1:2]))

LomanNJ_tib$exprs_head %<>% setNames(LomanNJ_tib$type)

cat("Per-study table. each row has a single assay")
LomanNJ_tib

cat("data format for each assay")
LomanNJ_tib$exprs_head

```







```{r}


```



```{r}


```
