# Curated_Metagenomics
Expanding the utility of curated metagenomics


# Motivation:
curatedMetagenomicsData is a unified, well annotated microbiome / metagenomics public data set.  
It contain 6 types of assay data for each study:  
    1. Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level  
    2. Presence of unique, clade-specific markers  
    3. Abundance of unique, clade-specific markers  
    4. Abundance of gene families  
    5. Metabolic pathway coverage  
    6. Metabolic pathway abundance  

Types 1-3 are generated by MetaPhlAn2; 4-6 are generated by HUMAnN2.  

Current popular object oriented classes in R for such data is phyloseq, however, it can handle only 1 omic data.  
This project will attempt to suggest some utility functions to transfer curatedMetagenomicsData into MAE, and introduce some simple analytics functions that are available on phyloseq.  

# Tasks:
1. Data wrangling conversion:  
  curatedMetagenomicData -> MAE.  

2. Analytic method (proof of concept):   
  Choose some phyloseq common functions that don't depends on the phylogeny tree slot, and have them for MAE input.  
  Expend to methods with more than one assay.  
  Fine resolution analysis: by taxonomy information embeded in assay 'analyte' names. 
  (Kingdom / Phylum / Class / Order / Family / Genus / Species / Strain)

3. Meta-analysts:
  Demonstrate analysis for multiple studies, and compare results.  


# Gaps:
Unlike phyloseq, we are not currently sure how taxonomy phylogeny tree can be integrated into MAE.  



![MultiAssayExperiment](https://bioconductor.github.io/BiocWorkshops/Ramos_MultiAssayExperiment/MultiAssayExperiment.png)

![phyloseq](https://journals.plos.org/plosone/article/figure/image?size=large&id=10.1371/journal.pone.0061217.g003)




# Introduction on data wrangling from curatedMetagenomicData into MAE:
(https://htmlpreview.github.io/?https://github.com/NCBI-Hackathons/Curated_Metagenomics/blob/master/curatedMetagenomicData_data_review.html)